public/app/features/alerting/unified/testSetup/plugins.ts

```diff
@@ -0,0 +1,102 @@
+import { HttpResponse, RequestHandler, http } from 'msw';
+
+import { PluginMeta, PluginType } from '@grafana/data';
+import { config } from '@grafana/runtime';
+
+export function setupPlugins(...plugins: PluginMeta[]): { apiHandlers: RequestHandler[] } {
+  const pluginsRegistry = new Map<string, PluginMeta>();
+  plugins.forEach((plugin) => pluginsRegistry.set(plugin.id, plugin));
+
+  pluginsRegistry.forEach((plugin) => {
+    config.apps[plugin.id] = {
+      id: plugin.id,
+      path: plugin.baseUrl,
+      preload: true,
+      version: plugin.info.version,
+      angular: plugin.angular ?? { detected: false, hideDeprecation: false },
+    };
+  });
+
+  return {
+    apiHandlers: [
+      http.get<{ pluginId: string }>(`/api/plugins/:pluginId/settings`, ({ params: { pluginId } }) => {
+        if (pluginsRegistry.has(pluginId)) {
+          return HttpResponse.json<PluginMeta>(pluginsRegistry.get(pluginId)!);
+        }
+        return HttpResponse.json({ message: 'Plugin not found, no installed plugin with that id' }, { status: 404 });
+      }),
```

Comment:
Would be nice if this logic went into the main `handlers` file that we now have?